tsk/img/vmdk.c

```diff
@@ -31,9 +31,8 @@ getError(libvmdk_error_t * vmdk_error,
     error_string[0] = '\0';
     retval = libvmdk_error_backtrace_sprint(vmdk_error,
         error_string, TSK_VMDK_ERROR_STRING_SIZE);
-    if (retval)
-        return 1;
-    return 0;
+    libvmdk_error_free(&vmdk_error);
+    return retval ? 1 : 0;
```

Comment:
This could also be `return retval != 0`;

Reply:
Done.